### PR TITLE
Add CronJob to OLM Dance custom-domains-operator

### PR DESCRIPTION
**Pull request by mjlshen**
[OSD-17042](https://issues.redhat.com//browse/OSD-17042)

This fixes #1793 which was not actually performing an OLM dance as intended

### CODE REVIEW

File: deploy/osd-17042-cdo-olm-dance/cronjob.yaml [hunk 1]
```diff
@@ -88,14 +91,15 @@ spec:
...
-              COUNT_SUB=$(oc -n "$NAMESPACE" get subscriptions.operators.coreos.com custom-domains-operator -o name | wc -l) || true
-              if [[ $COUNT_SUB -ge 1 ]]; then
-                oc patch subscriptions.operators.coreos.com -n "$NAMESPACE" custom-domains-operator --type json --patch='[ { "op": "remove", "path": "/metadata/annotations/kubectl.kubernetes.io~1last-applied-configuration" } ]'
+              # Check for the presence of CDO v0.1.201, then we know we're stuck
+              # stuck
+              CSV_FOUND=$(oc -n "$NAMESPACE" get clusterserviceversions.operators.coreos.com custom-domains-operator.v0.1.201-b9b8893 -o name --ignore-not-found) || true
+              if [[ ! -z "$CSV_FOUND" ]]; then
+                # It is present, so wipe out CDO
+                oc -n "$NAMESPACE" delete clusterserviceversions.operators.coreos.com -l operators.coreos.com/custom-domains-operator.openshift-custom-domains-operator=""
+                oc -n "$NAMESPACE" delete installplan.operators.coreos.com --all
```
Comment (jharrington22) [inline deploy/osd-17042-cdo-olm-dance/cronjob.yaml:100]: We want to only delete installplans for CDO 

`oc get installplans -n openshift-custom-domains-operator -l operators.coreos.com/custom-domains-operator.openshift-custom-domains-operator=""`

**Review by jharrington22 — APPROVED** [review]
/lgtm

**Comment by mrbarge** [comment]
Just a warning with this approach that we encountered last time.. deleting the subscription and reapplying it results in a subscription that can no longer be clustersynd'd over as the last-applied-configuration annotation in it will confuse the cluster sync. It is better to either delete the subscription and let it be resynced naturally, or remove the annotation via jq when dumping it out to json.

**Comment by mjlshen** [comment]
Thanks so much for catching this - I couldn't recall the specifics from the MGO OLM dance. We'll update this CronJob on Monday (after recharge) to delete the subscription wholesale

**Comment by mrbarge** [comment]
#1772 was a PR to fix the busted-sub issue from last time for MGO, so it may be able to be similarly leveraged for CDO on Monday. 🙂

**Pull request merged**